src/query/remote/client.go

```diff
@@ -36,9 +38,44 @@ import (
 	"github.com/m3db/m3/src/query/storage/m3/consolidators"
 	"github.com/m3db/m3/src/query/ts/m3db"
 	"github.com/m3db/m3/src/query/util/logging"
+	xgrpc "github.com/m3db/m3/src/x/grpc"
 	"github.com/m3db/m3/src/x/instrument"
 
+	"github.com/uber-go/tally"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+)
+
+const (
+	initResultSize             = 10
+	healthCheckInterval        = 60 * time.Second
+	healthCheckTimeout         = 10 * time.Second
```

Comment:
seems quite high for a basic ping.

Reply:
True, although this could be from one side of the world to the other (if you run in regions truly distributed globally). Granted still would have to be a huge timeout, but plausible to hit 2s I believe. 

Do you think 5s or 2.5s or even lower is suitable?

Reply:
2 is probably ok (an order of magnitude more than 200ms). 5s is probably ok  to be safe